subsys/net/lib/coap/coap.c

```diff
@@ -130,7 +130,7 @@ static int parse_option(const struct coap_packet *cpkt,
 	opt_len = 1;
```

Comment:
context->frag == NULL && context->offset == 0 means reached end of fragments. Can be gracefully exit here. But 'return 0' means parse_options() will  increase opt_len +1. It thinks it encountered PAYLOAD MARKER.